src/MappingBar/ZoneSearch.tsx

```diff
@@ -1,31 +1,100 @@
-import React, { FC } from 'react';
+import clone from 'lodash/cloneDeep';
+import isEqual from 'lodash/isEqual';
+import startsWith from 'lodash/startsWith'; // lodash is faster than native implementation
+import React, { FC, useCallback, useRef, useState } from 'react';
 
 import { TextField } from '@material-ui/core';
 import Autocomplete from '@material-ui/lab/Autocomplete';
+import useEventListener from '@use-it/event-listener';
+
+import { DEFAULT_ZONE } from '../data/constants';
+import { ZoneLight } from './';
 
 interface ZoneSearchProps {
-  zoneNames: string[];
+  zoneList: ZoneLight[];
   label: string;
-  value: string | null;
-  update: (zone: string | null) => void;
+  value: ZoneLight;
+  update: (zone: ZoneLight) => void;
 }
+const filterZones = (zoneList: ZoneLight[], state: object) => {
+  const newZoneList: ZoneLight[] = [];
+  const inputVal: string = (state as any).inputValue.toLowerCase();
+
+  for (const z of zoneList) {
+    if (startsWith(z.value, inputVal)) {
+      newZoneList.push(z);
+    }
+  }
+
+  return newZoneList;
```

Comment:
```suggestion
  const inputVal: string = (state as any).inputValue.toLowerCase();
  const inputTerms = inputVal.split(' ');

  return inputTerms.reduce((list, term) => list.filter(i => i.name.toLowerCase().indexOf(term) >= 0), zoneList)
```

This would enable fuzzy search, try searching for "qii v"

Reply:
Not sure how I feel about this, but we can give it a shot and see how it works in practice. Zone names are annoying af

Reply:
This completely breaks what I tried to do here, so I did both. If mine doesn't result in anything, it'll try yours. The point is to be quick at entering data and the list is small enough that searching through the list both ways is negligible.